DataLayer/gradle/libs.versions.toml

```diff
@@ -2,7 +2,7 @@
 android-gradle-plugin = "8.4.1"
 androidx-activity = "1.9.0"
 androidx-compose-bom = "2024.05.00"
-androidx-lifecycle = "2.8.1"
```

Comment:
Worth a comment, otherwise we will likely revert.

Also I think maybe we should consider moving past this, by updating to beta since they are getting close to stable.

Reply:
I was getting a compatibility issue and this is why I moved to 2.7 but it's now solved, so I will revert this back